<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1345</width>
    <height>771</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>40</y>
      <width>971</width>
      <height>471</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetMaximumSize</enum>
     </property>
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="widgetResizable">
        <bool>false</bool>
       </property>
       <widget class="MainDisplay" name="mainDisplay">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>365</width>
          <height>417</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QScrollArea" name="scrollArea_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="widgetResizable">
        <bool>false</bool>
       </property>
       <widget class="MainDisplay" name="mainDisplay2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>364</width>
          <height>417</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>121</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Select image</string>
    </property>
    <widget class="QCheckBox" name="checkBoxOriginal">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>30</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Original</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxNoisy">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Noisy</string>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxDenoised">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>70</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Denoised</string>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QSlider" name="areaMinSlider">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>70</y>
      <width>221</width>
      <height>29</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::NoTicks</enum>
    </property>
   </widget>
   <widget class="QSlider" name="areaMaxSlider">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>120</y>
      <width>221</width>
      <height>29</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>50</y>
      <width>66</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Area min</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>100</y>
      <width>66</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Area max</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>979</x>
      <y>29</y>
      <width>281</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Area</string>
    </property>
    <widget class="QLabel" name="labelAreaMin">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>30</y>
       <width>56</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="labelAreaMax">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>80</y>
       <width>56</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>190</y>
      <width>281</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Contrast</string>
    </property>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>91</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Contrast min</string>
     </property>
    </widget>
    <widget class="QSlider" name="contrastMaxSlider">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>90</y>
       <width>221</width>
       <height>29</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="contrastMinSlider">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>40</y>
       <width>221</width>
       <height>29</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>91</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Contrast max</string>
     </property>
    </widget>
    <widget class="QLabel" name="labelContrastMin">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>30</y>
       <width>56</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="labelContrastMax">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>80</y>
       <width>56</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>520</y>
      <width>204</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>121</width>
      <height>0</height>
     </size>
    </property>
    <property name="text">
     <string>Compute graph multi-thread</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>550</y>
      <width>381</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Patchs definition</string>
    </property>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Division in width</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Division in height</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxNx">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>10</y>
       <width>60</width>
       <height>27</height>
      </rect>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBoxNy">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>40</y>
       <width>60</width>
       <height>27</height>
      </rect>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>550</y>
      <width>120</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Color space</string>
    </property>
    <widget class="QCheckBox" name="checkBoxRGB">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>RGB space</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxHSV">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>HSV space</string>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxSV">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>SV space + H</string>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>550</y>
      <width>181</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Ordering</string>
    </property>
    <widget class="QCheckBox" name="checkBoxComponent">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>161</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Component-wise</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBoxInverseComponent">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>101</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Inverse</string>
     </property>
     <property name="autoExclusive">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>525</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="inputMethodHints">
     <set>Qt::ImhNone</set>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="alignment">
     <set>Qt::AlignBottom|Qt::AlignLeading|Qt::AlignLeft</set>
    </property>
   </widget>
   <widget class="QPushButton" name="adaptiveArea">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>340</y>
      <width>201</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Adaptive filtering: area</string>
    </property>
   </widget>
   <widget class="QPushButton" name="adaptiveContrast">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>410</y>
      <width>201</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Adaptive filtering: contrast</string>
    </property>
   </widget>
   <widget class="QSlider" name="adaptiveAreaSlider">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>370</y>
      <width>221</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>99</number>
    </property>
    <property name="value">
     <number>99</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="labelAdaptiveArea">
    <property name="geometry">
     <rect>
      <x>1220</x>
      <y>350</y>
      <width>56</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QSlider" name="adaptiveContrastSlider">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>440</y>
      <width>221</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>99</number>
    </property>
    <property name="value">
     <number>99</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="labelAdaptiveContrast">
    <property name="geometry">
     <rect>
      <x>1220</x>
      <y>410</y>
      <width>56</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <zorder>groupBox_2</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>groupBox</zorder>
   <zorder>areaMinSlider</zorder>
   <zorder>areaMaxSlider</zorder>
   <zorder>label_2</zorder>
   <zorder>label_3</zorder>
   <zorder>groupBox_3</zorder>
   <zorder>pushButton_2</zorder>
   <zorder>groupBox_4</zorder>
   <zorder>groupBox_5</zorder>
   <zorder>groupBox_6</zorder>
   <zorder>progressBar</zorder>
   <zorder>adaptiveArea</zorder>
   <zorder>adaptiveContrast</zorder>
   <zorder>adaptiveAreaSlider</zorder>
   <zorder>labelAdaptiveArea</zorder>
   <zorder>adaptiveContrastSlider</zorder>
   <zorder>labelAdaptiveContrast</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1345</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuComponent_Graph">
    <property name="title">
     <string>Component Graph</string>
    </property>
    <addaction name="actionLoad_Image"/>
    <addaction name="actionSynthetic_Image"/>
    <addaction name="actionSave_result"/>
   </widget>
   <widget class="QMenu" name="menuProcessing">
    <property name="title">
     <string>Processing</string>
    </property>
    <addaction name="actionSimplification"/>
    <addaction name="actionDetection"/>
   </widget>
   <widget class="QMenu" name="menuAdd_noise">
    <property name="title">
     <string>Add noise</string>
    </property>
    <addaction name="actionImpulse_noise"/>
    <addaction name="actionGaussian_noise"/>
   </widget>
   <addaction name="menuComponent_Graph"/>
   <addaction name="menuProcessing"/>
   <addaction name="menuAdd_noise"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad_Image">
   <property name="text">
    <string>Load Image</string>
   </property>
  </action>
  <action name="actionSimplification">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Simplification</string>
   </property>
  </action>
  <action name="actionDetection">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Detection</string>
   </property>
  </action>
  <action name="actionSynthetic_Image">
   <property name="text">
    <string>Synthetic Image</string>
   </property>
  </action>
  <action name="actionImpulse_noise">
   <property name="text">
    <string>Impulse noise</string>
   </property>
  </action>
  <action name="actionGaussian_noise">
   <property name="text">
    <string>Gaussian noise</string>
   </property>
  </action>
  <action name="actionSave_result">
   <property name="text">
    <string>Save result</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MainDisplay</class>
   <extends>QWidget</extends>
   <header location="global">maindisplay.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
